src/ucp/core/ucp_am.c

```diff
@@ -182,7 +182,11 @@ ucp_am_rdesc_in_progress(ucp_recv_desc_t *desc, ucs_status_t am_cb_status)
         return 0;
     }
 
-    desc->flags &= ~UCP_RECV_DESC_FLAG_INCOMPLETE;
+    /* Receive operation is started or user wants to keep this data. Clear
+     * UCP_RECV_DESC_FLAG_DONT_RELEASE flag, so it can be safely released later
+     * by ucp_am_data_release() or by completion of the receive operation.
+     */
+    desc->flags &= ~UCP_RECV_DESC_FLAG_DONT_RELEASE;
```

Comment:
can we remove the flag outside this function?

Reply:
removed